Point sign up button at new form in app

nickclyde opened the pull request:
Once https://github.com/CDCgov/prime-simplereport/pull/2346 is merged, we'll just need to merge this PR to switchover to the new in app form!

nullflux reviewed (approved):
🔥 🎉 

LGTM!

**question (non-blocking):** Will you follow this PR with one to remove the obsolete static pages?

nickclyde commented:
@nullflux Good catch... I may as well go ahead and do that in this PR! commit incoming

nickclyde commented:
@nullflux As it turns out there were a couple other places in the site pointing at the old sign up form... so glad you brought that up. Updated those other pointers and removed the old form. Also removed the e2e tests in this repo since they only tested that form!

emmastephenson reviewed (approved):
Woohoo!!
